Build failure: texmaker

### Nixpkgs version

- Unstable (25.11)

### Steps to reproduce

`environment.systemPackages = [ pkgs.texmaker ]`

### Can Hydra reproduce this build failure?

Yes, Hydra can reproduce this build failure.

### Link to Hydra build job

https://hydra.nixos.org/build/310491572

### Relevant log output

```console
Running phase: qtPreHook
@nix { "action": "setPhase", "phase": "qtPreHook" }
Running phase: unpackPhase
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking source archive /nix/store/vr3n09gfxh1a4waach1p2l20j3i0nh2c-texmaker-6.0.1.tar.bz2
source root is texmaker-6.0.1
setting SOURCE_DATE_EPOCH to timestamp 1745407377 of file "texmaker-6.0.1/CMakeLists.txt"
Running phase: patchPhase
@nix { "action": "setPhase", "phase": "patchPhase" }
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
Running phase: configurePhase
@nix { "action": "setPhase", "phase": "configurePhase" }
cmake flags: -DCMAKE_FIND_USE_SYSTEM_PACKAGE_REGISTRY=OFF -DCMAKE_FIND_USE_PACKAGE_REGISTRY=OFF -DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON -DCMAKE_BUILD_TYPE=Release -DBUILD_TESTING=OFF -DCMAKE_INSTALL_LOCALEDIR=/nix/store/ir5m0m4y6xvm71vcp3ysjf7wjy2g5mrm-texmaker-6.0.1/share/locale -DCMAKE_INSTALL_LIBEXECDIR=/nix/store/ir5m0m4y6xvm71vcp3ysjf7wjy2g5mrm-texmaker-6.0.1/libexec -DCMAKE_INSTALL_LIBDIR=/nix/store/ir5m0m4y6xvm71vcp3ysjf7wjy2g5mrm-texmaker-6.0.1/lib -DCMAKE_INSTALL_DOCDIR=/nix/store/ir5m0m4y6xvm71vcp3ysjf7wjy2g5mrm-texmaker-6.0.1/share/doc/texmaker -DCMAKE_INSTALL_INFODIR=/nix/store/ir5m0m4y6xvm71vcp3ysjf7wjy2g5mrm-texmaker-6.0.1/share/info -DCMAKE_INSTALL_MANDIR=/nix/store/ir5m0m4y6xvm71vcp3ysjf7wjy2g5mrm-texmaker-6.0.1/share/man -DCMAKE_INSTALL_INCLUDEDIR=/nix/store/ir5m0m4y6xvm71vcp3ysjf7wjy2g5mrm-texmaker-6.0.1/include -DCMAKE_INSTALL_SBINDIR=/nix/store/ir5m0m4y6xvm71vcp3ysjf7wjy2g5mrm-texmaker-6.0.1/sbin -DCMAKE_INSTALL_BINDIR=/nix/store/ir5m0m4y6xvm71vcp3ysjf7wjy2g5mrm-texmaker-6.0.1/bin -DCMAKE_INSTALL_NAME_DIR=/nix/store/ir5m0m4y6xvm71vcp3ysjf7wjy2g5mrm-texmaker-6.0.1/lib -DCMAKE_POLICY_DEFAULT_CMP0025=NEW -DCMAKE_FIND_FRAMEWORK=LAST -DCMAKE_STRIP=/nix/store/x8mydcgbry214s802nzvy7fdljx404ym-gcc-wrapper-14.3.0/bin/strip -DCMAKE_RANLIB=/nix/store/x8mydcgbry214s802nzvy7fdljx404ym-gcc-wrapper-14.3.0/bin/ranlib -DCMAKE_AR=/nix/store/x8mydcgbry214s802nzvy7fdljx404ym-gcc-wrapper-14.3.0/bin/ar -DCMAKE_C_COMPILER=gcc -DCMAKE_CXX_COMPILER=g++ -DCMAKE_INSTALL_PREFIX=/nix/store/ir5m0m4y6xvm71vcp3ysjf7wjy2g5mrm-texmaker-6.0.1 -DQT_OPTIONAL_TOOLS_PATH=/nix/store/qld4r900hwnf2qq2pl6v1811zgpj4i51-qttools-6.10.0 -DQT_OPTIONAL_TOOLS_PATH=/nix/store/qld4r900hwnf2qq2pl6v1811zgpj4i51-qttools-6.10.0 -DPROTOC_EXE=/nix/store/izjrnd79npwzbdcjiv0i36dcw3kigny9-protobuf-32.1/bin/protoc -DProtobuf_PROTOC_EXE=/nix/store/izjrnd79npwzbdcjiv0i36dcw3kigny9-protobuf-32.1/bin/protoc -DProtobuf_PROTOC_EXECUTABLE=/nix/store/izjrnd79npwzbdcjiv0i36dcw3kigny9-protobuf-32.1/bin/protoc
-- The CXX compiler identification is GNU 14.3.0
-- The C compiler identification is GNU 14.3.0
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /nix/store/x8mydcgbry214s802nzvy7fdljx404ym-gcc-wrapper-14.3.0/bin/g++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /nix/store/x8mydcgbry214s802nzvy7fdljx404ym-gcc-wrapper-14.3.0/bin/gcc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
-- Found Threads: TRUE
-- Found OpenGL: /nix/store/as2rg79vk4z7frb7hy27nmadc3fp1a4j-libglvnd-1.7.0/lib/libOpenGL.so
-- Found WrapOpenGL: TRUE
-- Found WrapVulkanHeaders: /nix/store/icyb0ffm7i4sajzphnf818firkqlx71h-vulkan-headers-1.4.328.0/include
-- Performing Test HAVE_STDATOMIC
-- Performing Test HAVE_STDATOMIC - Success
-- Found WrapAtomic: TRUE
-- Found Cups: /nix/store/d6spmgs7bvam7k32p15yx2wicr7cdf06-cups-2.4.14-lib/lib/libcups.so (found version "2.4.14")
-- Configuring done (4.5s)
[31mCMake Error at CMakeLists.txt:1446 (target_link_libraries):
  Target "texmaker" links to:

    Qt6::CorePrivate

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.


[0m
-- Generating done (0.4s)
[33mCMake Warning:
  Manually-specified variables were not used by the project:

    BUILD_TESTING
    CMAKE_EXPORT_NO_PACKAGE_REGISTRY
    CMAKE_POLICY_DEFAULT_CMP0025
    PROTOC_EXE
    Protobuf_PROTOC_EXE
    Protobuf_PROTOC_EXECUTABLE

[0m
[0mCMake Generate step failed.  Build files cannot be regenerated correctly.[0m
```

### Additional context

_No response_

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.17.3, NixOS, 25.11 (Xantusia), 25.11.20251019.5e2a59a`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.31.2`
 - channels(root): `"nixos"`
 - nixpkgs: `/nix/store/7agp54mgffm9m1wc1kgmkm37pvy18qhf-source`

### Notify maintainers

@Chaddai @markuskowa

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/applications/editors/texmaker/default.nix
+++ b/pkgs/applications/editors/texmaker/default.nix
@@ -22,6 +22,11 @@ stdenv.mkDerivation rec {
     hash = "sha256-uMI13wzY/XcUzXDTte42MWOwJUqd6pGAeBuPDi5GyvY=";
   };
 
+  patches = [
+    # Check if the patch can be removed next release
+    ./fix-build-with-qt-6-10.patch
+  ];
+
   buildInputs = [
     poppler
     qtbase

--- a/pkgs/applications/editors/texmaker/fix-build-with-qt-6-10.patch
+++ b/pkgs/applications/editors/texmaker/fix-build-with-qt-6-10.patch
@@ -1,0 +1,13 @@
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index ef6a4e6..b564be8 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -386,7 +386,7 @@ set(CMAKE_CXX_STANDARD 17)
+ set(CMAKE_CXX_STANDARD_REQUIRED ON)
+ set(CMAKE_AUTORCC ON)
+ 
+-find_package(Qt6 REQUIRED COMPONENTS Gui Concurrent Core Core5Compat Network PrintSupport Qml Widgets Xml LinguistTools)
++find_package(Qt6 REQUIRED COMPONENTS Gui Concurrent Core CorePrivate Core5Compat Network PrintSupport Qml Widgets Xml LinguistTools)
+ 
+ qt_standard_project_setup()
+ 

